### ISSUE TITLE
Bring back the railroad

### ISSUE THREAD

**Issue by ShadesOT** [We-the-People-civ4col-mod/Mod#98]
I think TAC had railroads (there are pages in the colopedia) and RAR labeled them carriages.

Railroads are a great idea. I would be great to have them back. But not just relabeled. Completely overhauled. Maybe just when there is a tech tree, or after a certain in game year. Very expensive compared to the other land vehicles and with a lot of prequisites. But very fast and capable to transport a lot of cargo and settlers. Good to tackle transportation needs in huge nations.

Like in real history. Railroads changed the flow of goods and people profoundly.

**Comment by albucctestuser** [comment]
Test for non-member using giving their suggestions.

**Comment by abmpicoli** [comment]
I think we could make it like a "build factory" event... A railroad can only be created next to another railroad. And when a factory is built at the city the city plot receives a railroad improvement.

**Comment by Nightinggale** [comment]
CivEffects #93 can be used to unlock units and routes. It would be a minor addition to the code to make Eras provide a CivEffect to whoever reach the era in question.

Regarding tech tree, #94 will aim at providing one for other mods. We should consider if we want to use it in RaRE and if so, how it should be used.

And yes I do want railroads back. They were reskinned at some point, but RaR had railroads when I joined. The argument was "no tech tree means you can build them in the 16th century" and it was apparently not debatable.

Regarding trains, we should think of a new and clean way to implement them in the DLL. Currently the train identifier is (land based && 6 cargo slots). It would be better to make an xml implementation of "required route" to units. Something generic, which could work for riverboats #34 too.

**Comment by ErikSandbraaten** [comment]
I think that expanding the game to cover the age of steam would be cool, but it would definitely require a tech tree in my opinion. I've always wanted to play an "Age of Imperialism" mod😄

**Comment by ShadesOT** [comment]
But ..... imo colonization always was about a game that had no steep learning curve, no tech tree, no stress to replace units with stronger, newer ones, etc. We should keep that in mind. To be honest, I pretty much oppose a tech tree. I fear, if we start going down that road, more and more ideas will pop up. until we have a fully fledged tech tree that spans from 1492 to 1900 or something.

At least we should have the option to switch it off. And regarding history, ... the golden age of railroads was after independence. So, I merely vote for a Time event for railroads and the large river mechanic for transporting goods. After all, cities were built on shores of oceans and large rivers, for exactly the reason that goods and people could be transportet to and from the city, before there were railroads. The hinterlands were just accessible by horse waggons, which was slow and painful.

Maybe we could map that to the game mechanics? lets think about it somewhen later.

**Comment by Nightinggale** [comment]
I'm preparing the DLL for other mods and part of the plan is to support a tech tree. However that's not the same as we need to use a tech tree in all mods. It's only an option.

The idea with CivEffects #93 is that it's a bunch of classes we can give to players. Ownership can then be set to turn railroad routes, railroad build and train units on and off (as well as lots of other stuff). The real question is trigger condition and while a tech tree is obvious, the idea is that we can set many different ways to assign. For this purpose I would propose era and/or events. We could make an event, which triggers for each player once that player has 10 cities in the same plotgroup #82, as in the player unlocks by the need due to having such a big road network.

In short we are dealing with two different questions here: should we have railroads in late game and if so, how are they unlocked?

`the golden age of railroads was after independence`
That's pretty much US only. A lot of colonies didn't gain independence until the mid 20th century. India has the most km/miles of track of all countries in the world and most of it was built by the British before independence. Railroads in Africa are also mainly colonial and while I'm not too much into South American independence history, I would assume a bunch of colonial railroads there too.

**Comment by ErikSandbraaten** [comment]
I agree that the tech tree should not be mandatory, If we ever add one, it should probably be in WTP 3.x or something like that. I am compeltely opposed to any time\turn unlock on railraids, if they're ever added they should be unlocked by a TT.

**Comment by ShadesOT** [comment]
is TT a Target Tech?

**Comment by ErikSandbraaten** [comment]
Tech Tree

**Comment by abmpicoli** [comment]
I think that tech trees are a totally new feature into the game. How will it be unlocked? A new resource? Bells? Then what would be the technologies? And then make the AI use the tech... 

I think that we don't need to fix a time, but a set of infrastructure requests... If we have a building (train station), and then only allow railroads to be created next to another railroad, we have an "implicit" tech tree here... that fits better the current game. I also think it will be easier to implement...

**Comment by ShadesOT** [comment]
Infrastructure requirements would be the way to go for me as well. Though I think the XML and game engine should allow for each of the 3 criteria. TT, time and infrastructure.

As for infrastructure requirement there could be the rail foundry, the railroad workshop with several tiers (which allow for more powerful trains) and (combined with #149 ) the reailroad engineer, which can only be commissioned in cities which have a rail foundry.
Not saying that it has to be like this, just catching the ideas that pop up in my head like mushrooms in dark, wet and warm places.

**Comment by Nightinggale** [comment]
`
Though I think the XML and game engine should allow for each of the 3 criteria. TT, time and infrastructure.
`
CivEffects #93 is the answer. What we do is add a CivEffect, which enables units, buildings and route building. Next is the trigger event for a player to gain that CivEffect and we can pick TT, eras, events and a lot of other options.
In other words CivEffects can already provide what you said and it's with features planned before this ticket was even opened. Using it for railroads is just a bonus. CivEffects are well underway in the CivEffect branch and CvPlayer already has canUseUnit() and canUseBuilding().

`
As for infrastructure requirement there could be the rail foundry, the railroad workshop with several tiers (which allow for more powerful trains) and (combined with #149 ) the railroad engineer, which can only be commissioned in cities which have a rail foundry.
`
So far the train has been a constructed unit, not a profession.

**Comment by ShadesOT** [comment]
> So far the train has been a constructed unit, not a profession.

I did not mention, that my idea for the railroad engineer is to be the only unit which can build rail tracks.

**Comment by Nightinggale** [comment]
I believe the word you are looking for is [navvy](https://en.wikipedia.org/wiki/Navvy). A railroad engineer is a US term for the person driving a train.

A navvy profession would make sense and since lots of navvies were needed, it would make sense to make railroad building slow, hence requiring lots of units gaining the profession in question. However we should consider historical accuracy vs gameplay balance and fun in this particular case. If railroads become too hard to build, they won't be used and if they are so good that they are worth such an investment, gaining railroads would be problematic from a game balance point of view.

CivEffects is planned to be able to turn professions on as well.

Maybe railroad connection between a colony and a mine could provide a production bonus. The early railroads were not meant for speed or transporting passengers (though they offered that). It was all about transport capacity as transport was bottlenecking both production and delivery to factories/consumers. A mine to city railroad would increase the flow of mined resources into the city if we do that part historical accurate.

**Comment by ShadesOT** [comment]
> Road transport were way too expensive in horse food for large scale transports. It explains the canal digging frenzy in mid 18th century England.

Tha also makes me think of another idea which probably is to complex to implement: food and coal for fueling transports like horse waggons and railroads and maybe even ironclads ( https://en.wikipedia.org/wiki/Ironclad_warship)

**Comment by Nightinggale** [comment]
`
What does that mean in this particular case?
`
CvUnit::canHaveProfession(eProfession) is used to tell if a unit can switch to a certain profession. CivEffects will add a check for owner->canUseProfession(eProfession). The answer to that is to return cache > 0. This cache is then the value of all owned CivEffects put together. This means a CivEffect adding 1 will enable a profession while a CivEffect providing -1 can disable a profession. It's familiar to anybody modding civ4 as it's the system used by civ4 techs.

`
food and coal for fueling transports like horse waggons and railroads and maybe even ironclads 
`
Micromanagement alert (again). However we could consider #146

**Comment by NoFreeUsernameDammit** [comment]
> One thing to consider is unlocking railroads in multiple steps. Ignoring the railroad in ancient Greece, the railroad car is decades older than the locomotive. Before the locomotive, everything was horse drawn, but to cut costs of horse food, people wanted to use as few horses as possible while transporting everything they needed to transport. One work horse could pull:
> 
> * wagon on road: 1 ton
> * wagons on rails: 10 tons
> * barge on canal: 40 or 60 tons depending on which source you trust
> 
> Road transport were way too expensive in horse food for large scale transports. It explains the canal digging frenzy in mid 18th century England.
> 
> In other words we could consider first generation train to be horses with horse speed, but have a lot of cargo slots, say 4-5 and moving with wagon train speed.

Isn't that pretty much what current system is? Although even now, it's too fast and too high-capacity, compared to sea transport. After all, carriages can carry as much cargo as galleon which would be somewhere in 500ish ton range, since ingame galleons fill in role of largest naval transport of the time, and travel at speed 12 tiles/turn, more than twice as much as sailing ship which would usually move at steady 5-10 knots per hour (ballpark 9-18 kilometers per hour).

**Comment by ShadesOT** [comment]
> After all, carriages can carry as much cargo as galleon

Yes, this is a big  inbalance. It stems from carriages being relabeled and reskinned trains and plastered country roads being relabeled railway tracks. TAC (the ancestor of RAR) had railroads and the RAR crew relabeled and reskinned them into carriages.

**Comment by Nightinggale** [comment]
You are right. We need to not just add railroads, but rethink the entire transport system for balance. However it sounds right that steam powered railroads were faster. When the Liverpool to Manchester railroad opened in 1830 (the first railroad, which was something we would consider a railroad), one of the owners was hit by a train. He was put on a locomotive and it moved at full speed to Manchester to get him to the doctors, though he died during the next night. This was both the first machine powered ambulance transport and the world speed record of 40 mph or 64 km/h. While disputed at the time, modern remakes can exceed 50 km/h and nobody dares to go flat out for safety reasons, meaning that claimed speed is looking likely, but we don't know for sure. America followed suit the next year with the John Bull locomotive, which has reached 35 mph. Railroads started out fast.

**Comment by NoFreeUsernameDammit** [comment]
Yeah, but that's early 19th century. But the game actually ends with year 1800, and victory is usually achieved no later than 1730's, and by 1650's you usually have carriage system up and running...so it makes no sense in making steam power available ingame.

**Comment by ShadesOT** [comment]
Just dropping some ideas:
- Different route categories: #188 
- Building rail track needs a depletable yield (rails), which must be carried by the unit building the track ( see #141)
- **Combination of unlock by tech tree and buildings**: Profession "railroad architect" can be placed in railroad related building (eg the engine workshop/the inventors workshop/the railroad headquarters) where it produces railroad invention points, which lead to the invention of railroad related stuff (basic rails&track, several tiers of engines (horse powered, steam powered 1, steam powered 2), tiers of etc ...). The important difference to an overall techtree is, that the invention points are railroad specific.

**Comment by raystuttgart** [comment]
The big discussion why Railroads was removed was about historical accuracy for that period of time.
It is absolutely true that during the time period that this mod (RaR - now WTP) coverst they simply did not exist. That is a fact we cannot argue about.

TAC never had Railroads because of that reasons even though it was discussed there as well.
(Trains were actually requested in TAC quite often because people liked the mechanics from CIV4.)

Railroads ended up in RaR because I had implemented that feature in my prototype for a SCI-FI offspin or RaR (Path to the Throne, which was Dune inspired). 
I basically had all freedom to implement whatever concept I wanted because it was SCI-FI.

But I then was convinced again by Schmiddie that our (or at least many) Players did not like it and thus it was removed or to be accurate actually replaced.
We found the consense to keep the game mechanics and just redo the graphics and texts ("Plastered Roads" instead of "Railroads", "Carriage" instead of "Train")
That was good enough for me.

If we ever build a fully developed TechTree we could build Trains back in as a very very late Tech.
That would actually be awesome. From my perspective a **little** bit of "alternative history" is always nice. Civ4Col does not need to be a 100% realistic game.

But without a Tech mechanic let us please not do it.
1) I do not want to be discussing this again and again with complaining people from community.
2) In the community it would look very stupidly if we revert our decisions again and again without any reasonable explanations. That is kind of "unprofessional" - even if we are just modding for fun.

**Comment by raystuttgart** [comment]
Already working on it at the moment.
Thus it can be closed as issue now.
